src/PureScript/Backend/Optimizer/Convert.purs

```diff
@@ -443,19 +443,26 @@ toBackendExpr = case _ of
           makeLet Nothing (toBackendExpr expr) \tmp ->
             next (Array.snoc idents tmp)
       )
-      ( \idents ->
-          toInitialCaseRows idents alts \caseRows ->
+      ( \idents -> do
+          { dataTypes, implementations } <- ask
+          toInitialCaseRows dataTypes implementations idents alts \caseRows ->
             buildCaseTreeFromRows caseRows
       )
       exprs
       []
   where
-  toInitialCaseRows :: Array Level -> Array (CaseAlternative Ann) -> (Array CaseRow -> ConvertM BackendExpr) -> ConvertM BackendExpr
-  toInitialCaseRows idents alts useCaseRowsCb =
+  toInitialCaseRows
+    :: Map ProperName DataTypeMeta
+    -> BackendImplementations
```

Comment:
I think we can just keep pass the ConvertEnv around (or go through ConvertM) rather than pulling specific fields out of it and threading those around.

Reply:
Args are now passed implicitly via `ConvertM`.